python/dlr/tflite_model.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import os
+import tensorflow.lite as lite
+from .api import IDLRModel
+
+
+def _get_input_and_output_names(intrp):
+    """
+    Get input and output tensor names
+
+    Parameters
+    ----------
+    intrp : :py:class:`str`tf.lite.Interpreter`
+        TFLite Interpreter
+
+    Returns
+    -------
+    input_tensor_map : Input tensor Map which maps tensor_name to tensor_id
+    output_tensor_list : Output tensor list. List element is pair (name, id)
+    """
+    input_tensor_map = {}
+    output_tensor_list = []
+    for d in intrp.get_input_details():
+        input_tensor_map[d["name"]] = d["index"]
+    for d in intrp.get_output_details():
+        output_tensor_list.append((d["name"], d["index"]))
+    return input_tensor_map, output_tensor_list
+
+
+class TFLiteModelImpl(IDLRModel):
+    """
+    TFLiteModelImpl is a wrapper on top of Tensorflow Lite which implements IDLRModel API
+
+    Parameters
+    ----------
+    tflite_file : :py:class:`str`
+        Full path to TFLite file (.tflite file)
+    dev_type : :py:class:`str`
+        Not Supported
+    dev_id : :py:class:`int`
+        Not Supported
+    """
+    def __init__(self, tflite_file, dev_type=None, dev_id=None):
+        if not tflite_file.endswith(".tflite"):
+            raise ValueError("Not a TFLite file: {}".format(tflite_file))
+        if not os.path.exists(tflite_file):
+            raise ValueError("TFLite file {} doesn't exist".format(tflite_file))
+        if dev_type is not None:
+            raise NotImplementedError("dev_type parameter is not supported")
+        if dev_id is not None:
+            raise NotImplementedError("dev_id parameter is not supported")
+
+        logging.info("Loading TFLite file: {}".format(tflite_file))
+        self._intrp = lite.Interpreter(model_path=tflite_file)
+        self._intrp.allocate_tensors()
+        self._input_tensor_map, self._output_tensor_list = _get_input_and_output_names(self._intrp)
+
+    def _validate_input_name(self, name):
+        if name not in self._input_tensor_map:
+            raise ValueError(
+                "Invalid input tensor name '{}'. List of input tensor names: {}".format(name, self.get_input_names()))
+
+    def _validate_input(self, input_values):
+        if isinstance(input_values, dict):
+            for k in input_values.keys():
+                if not isinstance(k, str):
+                    raise ValueError("input key must be string")
+                self._validate_input_name(k)
+        else:
+            raise ValueError("input_values must be of type dict")
+
+    def get_input_names(self):
+        """
+        Get all input names
+
+        Returns
+        -------
+        out : list of :py:class:`str`
+        """
+        names = list(self._input_tensor_map.keys())
+        names.sort()
```

Comment:
What's the original oder? Do we need to sort them, or we can make sorting optional to the users (i.e. `get_input_name(self, sorted=False)`?

Reply:
removed `names.sort()` and use OrderedDict for `_input_tensor_map` instead